src/main/scala/it/unibo/authsim/library/dsl/cryptography/algorithm/hash/HashFunction.scala

```diff
@@ -0,0 +1,120 @@
+package it.unibo.authsim.library.dsl.cryptography.algorithm.hash
+
+import com.google.common.hash.Hashing
+import com.google.common.io.BaseEncoding
+import it.unibo.authsim.library.dsl.cryptography.algorithm.CryptographicAlgorithm
+import org.apache.commons.codec.binary.Hex
+import org.apache.commons.codec.digest.DigestUtils
+
+import java.nio.charset.StandardCharsets
+
+/**
+ * Trait representing the hash function
+ */
+trait HashFunction extends CryptographicAlgorithm:
+  /**
+   * Method that perform the hash conversion: it converts the input value into a string created according
+   * to the hash algorithm chosen for the conversion.
+   *
+   * @param value             input value to transfrom
+   * @tparam A                type value of the input
+   * @return                  a string representing the hashed input value
+   */
+  def hash[A](value: A): String
+
+  /**
+   * Setter for the salt value
+   *
+   * @param salt              a new salt to use during the hash transformation
+   * @tparam A                type value of the input
+   */
+  def _salt[A](salt:A): Unit
```

Comment:
I don't get why this two methods have a generic type `A`, as hash functions use strings as input.

Should they accept only `String`s?